src/file.cpp

```diff
@@ -946,21 +981,21 @@ namespace libtorrent
 	{
 #if defined TORRENT_WINDOWS && !defined TORRENT_MINGW
 #if TORRENT_USE_WSTRING
-		wchar_t cwd[TORRENT_MAX_PATH];
-		_wgetcwd(cwd, sizeof(cwd) / sizeof(wchar_t));
+		wchar_t* cwd { ::_wgetcwd(nullptr, 0) };
```

Comment:
yes, it's allocate by std::malloc and must be free after usage. according msdn examples it's a proper way to retrieve cwd > MAX_PATH